includes/class-admin.php

```diff
@@ -121,6 +134,33 @@ public static function register_settings() {
 		);
 	}
 
+	/**
+	 * Update ActivityPub plugin
+	 */
+	public static function version_check() {
+		if ( ! function_exists( 'get_plugin_data' ) ) {
+			require_once( ABSPATH . 'wp-admin/includes/plugin.php' );
+		}
+		$plugin_data = \get_plugin_data( ACTIVITYPUB_PLUGIN );
+		$activitypub_db_version = \get_option( 'activitypub_version' );
+
+		// Needs update
+		if ( empty( $activitypub_db_version ) || \version_compare( $plugin_data['Version'], $activitypub_db_version, '>' ) ) {
+			// Check for specific migrations
+
+			if ( version_compare( '0.13.5', $activitypub_db_version, '>' ) ) {
+				// This updates post_meta with _activitypub_permalink_compat.
+				// Posts that have this meta will be backwards compatible with their permalink based ActivityPub ID (URI)
+
+				// This may create false positives, where the permalink has changed (slug, permalink structure) since federation,
```

Comment:
@pfefferle Do you think this is a sufficient upgrade/migration plan? 

1. On update, Mark all existing activitypub post types with a post meta: `_activitypub_permalink_compat` of the existing permalink.

2. In the tools page, a List_Table displays all these marked posts, a column shows how many comments are associated with the post. The following text would alert the user:

> Posts with ActivityPub Comments, should be treated with care, existing interactions in the fediverse but not on site will be lost. 

2. a) I think I should add some more alerting style to the rows with comments. 
2. b) Row actions allow Deleting the post on the fediverse, or Deleting and Re-sharing (Announce/Boost) the original post with the new ID (this removes existing interactions from the fediverse)

Any thoughts, on this flow, or the language?